### PR TITLE
Stop using old API to get distribution and bucket db.

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

File: storage/src/tests/distributor/bucketdbupdatertest.cpp [hunk 1]
```diff
@@ -1835,10 +1841,11 @@ BucketDBUpdaterTest::mergeBucketLists(
     }
 
     BucketDumper dumper(includeBucketInfo);
-    getBucketDBUpdater().getDistributorComponent()
-            .getBucketDatabase().forEach(dumper);
-    getBucketDBUpdater().getDistributorComponent()
-            .getBucketDatabase().clear();
+    auto &bucketSpaceRepo(getBucketDBUpdater().getDistributorComponent().getBucketSpaceRepo());
+    auto &distributorBucketSpace(bucketSpaceRepo.get(makeBucketSpace()));
```
Comment (vekterli) [inline storage/src/tests/distributor/bucketdbupdatertest.cpp]: These common lines could be factored out into a `defaultDistributorBucketSpace()` or similarly named test member function

**Review by vekterli — APPROVED** [review]
LGTM, see comment on test code de-duping

**Review by vekterli — APPROVED** [review]
:+1:

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**